buildenv/jenkins/openjdk_tests

```diff
@@ -302,6 +302,11 @@ timestamps{
                 LABEL += "&&!sw.os.cent.7"
             }
 
+            // JDK23+ isn't supported on machines prior to Mac 11 https://github.com/eclipse-openj9/openj9/issues/19694
+            if (params.JDK_IMPL == "openj9" && PLATFORM == "x86-64_mac" && params.JDK_VERSION && params.JDK_VERSION.toInteger() >= 23) {
```

Comment:
When this is used with jdknext, is `JDK_VERSION` "24" or "next"?

Reply:
Sure, but the problem here will be if `JDK_VERSION` is "next", that `toInteger()` will throw a `NumberFormatException`.

Reply:
I think we will get NumberFormatException when JDK_VERSION = next. I will fix this.

Reply:
That test won't answer the question: it is for aarch64_mac instead of x86-64_mac.

In https://openj9-jenkins.osuosl.org/job/Test_openjdknext_j9_sanity.openjdk_aarch64_mac_Personal/9/console; I see
```
14:41:00  LEVELS: [sanity]
14:41:00  JDK_VERSIONS: [8]
14:41:00  GROUPS: [openjdk]
14:41:00  ARCH_OS_LIST: [aarch64_mac]
```
The `JDK_VERSIONS: [8]` part was a surprise. Does that make sense to anyone else?

Reply:
The test pipeline cannot consume `next`, so it will try to get Java version and set it at a later stage.

Reply:
But where did "8" come from in a job testing Java 24 (jdknext)?

Reply:
No related problem in that build. Afaik, `JDK_VERSION` is not set to next.

Reply:
It didn't get that that test because `PLATFORM == "x86-64_mac"` was not true.

Reply:
https://openj9-jenkins.osuosl.org/job/Pipeline-Build-Test-Personal/618/

Reply:
ok, with the correct platform we get NumberFormatException
```
java.lang.NumberFormatException: For input string: "next"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Integer.parseInt(Integer.java:652)
	at java.base/java.lang.Integer.valueOf(Integer.java:983)
	at org.codehaus.groovy.runtime.StringGroovyMethods.toInteger(StringGroovyMethods.java:3319)
```